### ISSUE TITLE
CLI: Verify

### ISSUE THREAD

**Issue by jesperpedersen** [pgmoneta/pgmoneta#267]
Create a `pgmoneta-cli verify <server> <id>` command that verifies the manifest against the files on-disk

**Comment by jesperpedersen** [comment]
This could be expanded to run in the background as well with a default of `0` aka off

**Comment by iuryr** [comment]
Hi @jesperpedersen. This message is just to say that I began working on this issue.

**Comment by jesperpedersen** [comment]
@iuryr #266 have been merged

**Comment by iuryr** [comment]
Jesper,

just to clarify, this feature aims to verify the checksum of each file in the manifest file generated by pgmoneta against the files 'generated' by pgmoneta backup, right?

**Comment by jesperpedersen** [comment]
@iuryr Yes, you have to go through the `backup_manifest` file, and verify that the on-disk files still have the same SHA-256 value

**Comment by jesperpedersen** [comment]
@iuryr Still ok on this ?

**Comment by iuryr** [comment]
Hi @jesperpedersen ! Still ok. Took me more than I expected to understand the communication between pgmoneta and pgmoneta-cli processes. Now I'm working on the feature itself. Maybe I can submit a PR by the end of next week. Let me know if this issue is blocking somebody

**Comment by jesperpedersen** [comment]
@iuryr No, you are ok - you can always update the issue or discussion with your thoughts and progress

**Comment by iuryr** [comment]
> This could be expanded to run in the background as well with a default of `0` aka off

Hi @jesperpedersen , in my implementation I'm running the checking in a fork() of pgmoneta main process. Is this what you mean by "run in the background"? Also, can you explain what you mean my a default of `0`?

Finally, a sucessfull check (meaning the on-disk files checksums match the checksum reported in backup_manifest should make pgmoneta-cli print a message to the terminal? Or should it print something in case it does not match? Or another option?

**Comment by jesperpedersen** [comment]
Yes, you should run it in a `fork()`, and you can use the worker pool as well.

Just do the explicit `verify` command first, then we can look at doing automatic verification in the background - you may want to follow along on https://github.com/pgmoneta/pgmoneta/discussions/312

Yes, there should be some sort of outout - also in JSON format. All files that doesn't match must be listed.

Hope this is helpful

**Comment by Jubilee101** [comment]
Hey @iuryr , note that we already have [manifest verfication](https://github.com/pgmoneta/pgmoneta/blob/5b0bb61487efc0b927289cec49ddeddbe00beab4/src/libpgmoneta/manifest.c#L51), so maybe you could base your work on that to save some trouble.

**Comment by iuryr** [comment]
Thank you both for the clarification and tips. I'll get back to you with the PR in a couple of days, then we can continue a more focused discussion.

**Comment by jesperpedersen** [comment]
@iuryr Still ok ?

**Comment by iuryr** [comment]
Hi @jesperpedersen , still ok! Had some trouble with messages between cli and main process, but I think I'm on the right track now. Sorry its taking longer than anticipated. Should force push another commit on the PR the next few days

**Comment by jesperpedersen** [comment]
@iuryr Just let Haoran or I know if you have questions

**Comment by iuryr** [comment]
@jesperpedersen Just reported the situation on the draft PR
